go/mysql/flavor_mysql.go

```diff
@@ -430,7 +430,7 @@ const TablesWithSize80 = `SELECT t.table_name,
 		SUM(i.allocated_size)
 	FROM information_schema.tables t
 		LEFT JOIN information_schema.innodb_tablespaces i
-	ON i.name LIKE CONCAT(t.table_schema, '/', t.table_name, IF(t.create_options <=> 'partitioned', '#p#%', '')) COLLATE utf8mb3_general_ci
+	ON SUBSTRING_INDEX(i.name, '#p#p', 1) = CONCAT(t.table_schema, '/', t.table_name) COLLATE utf8mb3_general_ci
```

Comment:
`#p#p` is only true if partitions are named as `p0, p1, etc.` but not if they're named `foo1, foo2, f3`. I'm checking if all that's needed is to change into `#p#`, otherwise momentarily marking as "request changes" so the PR isn't merged yet.

Reply:
Good catch! I actually thought the tablespace names were always 0,1,2... but you're right:
```
mysql> CREATE TABLE trb3 (id INT, name VARCHAR(50), purchased DATE)     PARTITION BY RANGE( YEAR(purchased) ) (         PARTITION my1990 VALUES LESS THAN (1990),         PARTITION my1995 VALUES LESS THAN (1995),         PARTITION p2 VALUES LESS THAN (2000),         PARTITION p3 VALUES LESS THAN (2005)     );
Query OK, 0 rows affected (0.06 sec)

mysql> select name from information_schema.innodb_tablespaces where name like "vt_commerce/trb3%";
+---------------------------+
| name                      |
+---------------------------+
| vt_commerce/trb3#p#my1990 |
| vt_commerce/trb3#p#my1995 |
| vt_commerce/trb3#p#p2     |
| vt_commerce/trb3#p#p3     |
+---------------------------+
4 rows in set (0.03 sec)
```